src/CMakeLists.txt

```
@@ -18,6 +18,10 @@ if (ENABLE_FORTRAN)
    enable_language(Fortran)
 endif()
 
+if (AXOM_ENABLE_TESTS)
+    set(ENABLE_GMOCK ON CACHE STRING "")
+endif()
```

I don't think `AXOM_ENABLE_TESTS` exists yet here.
It is created when loading in the AxomOptions below.

If you enable tests with `ENABLE_TESTS`, then the following *should* work:
```suggestion
if (ENABLE_TESTS)
    set(ENABLE_GMOCK ON CACHE STRING "")
endif()
```

---

@white238 -- is this where the logic for enabling GMOCK should go?

Do we want to optionally enable GMOCK and guard against it?

---

Here's where `AXOM_ENABLE_TESTS` is created: 
* AxomOption: https://github.com/LLNL/axom/blob/858240e755466245b27d102c61bfcda480e26214/src/cmake/AxomOptions.cmake#L23

Which is called from:
* CMakeBasics: https://github.com/LLNL/axom/blob/858240e755466245b27d102c61bfcda480e26214/src/cmake/CMakeBasics.cmake#L10
* CMakeLists: https://github.com/LLNL/axom/blob/858240e755466245b27d102c61bfcda480e26214/src/CMakeLists.txt#L102

---

yes `ENABLE_GMOCK` has to be enabled before BLT is loaded.

---

I had tested with `ENABLE_GMOCK` already set in my cache, which caused things to work locally, but not when I pushed.

It looks like the best options might be to just set `ENABLE_GMOCK` unconditionally (where I currently have it, or further down before BLT). Is that right? I don't see a better option, and it shouldn't hurt anything.

---

You should not be enabling gmock unconditionally. It should be guarded by ENABLE_TESTS because its only used there.